data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -94,19 +96,55 @@ public S3SelectObjectWorker(final S3ObjectRequest s3ObjectRequest) {
 
     public void parseS3Object(final S3ObjectReference s3ObjectReference, final AcknowledgementSet acknowledgementSet) throws IOException {
         try{
-            selectObjectInBatches(s3ObjectReference, acknowledgementSet);
+            selectObject(s3ObjectReference, acknowledgementSet);
         } catch (Exception e){
             LOG.error("Unable to process object reference: {}", s3ObjectReference, e);
             s3ObjectPluginMetrics.getS3ObjectsFailedCounter().increment();
             throw new IOException(e);
         }
     }
 
-    private void selectObjectInBatches(final S3ObjectReference s3ObjectReference, final AcknowledgementSet acknowledgementSet) throws IOException {
+    private void selectObject(final S3ObjectReference s3ObjectReference, final AcknowledgementSet acknowledgementSet) throws IOException {
         final InputSerialization inputSerialization = getInputSerializationFormat(serializationFormatOption);
-        InputStream inputStreamList;
         final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
 
+        if (isScanRangeSupported()) {
```

Comment:
It's unfortunate that we can't have a scan range for some data types, but for my own understanding is there a difference in performance here between using scan range or just input stream?

Reply:
I'm not sure. I saw very similar performance between normal S3 source and S3 select with batching so I think the difference is minimal